app/main.py

```diff
@@ -1,3 +1,74 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: float) -> Distance:
```

Comment:
The `__add__` method should handle addition with both `Distance` instances and numeric types. The current implementation only checks if `other` is not a `Distance`, but it should also ensure `other` is a numeric type (e.g., `int` or `float`) when it's not a `Distance`.